src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,65 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (isset($options['extra']['curl']) && \is_array($options['extra']['curl'])) {
+            $extraCurlopts = array_keys($options['extra']['curl']);
```

Comment:
no need for this line, let's iterate over `$options['extra']['curl']` directly

Reply:
fixed.